File: PegionClocking/SMSWindowService/DAL/SMSDal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Data;
using System.Data.SqlClient;
using SMSWindowService.Config;

namespace SMSWindowService.DAL
{
    public class SMSDal
    {
        #region Variable
        DatabaseConnection dbconn;
        #endregion

        public String ActivationCode { get; set; }
        public String ModemID { get; set; }

        public void InboxSave(String SMSID, String SMSContent, String Sender, String SMSDate, String SMSTime, String ModemIDValue)
        {
            try
            {
                //string message = "";

                DataSet dataResult = new DataSet();
                dbconn = new DatabaseConnection();
                dbconn.DatabaseConn("InboxSave", "Local");

                if (dbconn.sqlConn.State == ConnectionState.Open) dbconn.sqlConn.Close();
                dbconn.sqlConn.Open();
                dbconn.sqlComm.Parameters.Clear();
                dbconn.sqlComm.Parameters.AddWithValue("@SMSID", SMSID);
                dbconn.sqlComm.Parameters.AddWithValue("@SMSContent", SMSContent);
                dbconn.sqlComm.Parameters.AddWithValue("@Sender", Sender);
                dbconn.sqlComm.Parameters.AddWithValue("@SMSDate", SMSDate);
                dbconn.sqlComm.Parameters.AddWithValue("@SMSTime", SMSTime);
                dbconn.sqlComm.Parameters.AddWithValue("@ActivationCode", ActivationCode);
                dbconn.sqlComm.Parameters.AddWithValue("@ModemID", ModemID);
                dbconn.sqlComm.ExecuteNonQuery();
                dbconn.sqlConn.Close();

                //return message;
            }
            catch (Exception ex)
            {

                throw ex;
            }
        }

        public void OutboxSave(String SMSContent, String Recipient, String Status, String Remarks, Int64 InboxID, String InboxType)
        {
            try
            {
                DataSet dataResult = new DataSet();
                dbconn = new DatabaseConnection();
                dbconn.DatabaseConn("OutboxSave", "Main");

                if (dbconn.sqlConn.State == ConnectionState.Open) dbconn.sqlConn.Close();
                dbconn.sqlConn.Open();
                dbconn.sqlComm.Parameters.Clear();
                dbconn.sqlComm.Parameters.AddWithValue("@SMSContent", SMSContent);
                dbconn.sqlComm.Parameters.AddWithValue("@Recipient", Recipient);
                dbconn.sqlComm.Parameters.AddWithValue("@Status", Status);
                dbconn.sqlComm.Parameters.AddWithValue("@StatusRemarks", Remarks);
                dbconn.sqlComm.Parameters.AddWithValue("@InboxID", InboxID);
                dbconn.sqlComm.Parameters.AddWithValue("@InboxType", InboxType);
                dbconn.sqlComm.ExecuteNonQuery();
                dbconn.sqlConn.Close();

            }
            catch (Exception ex)
            {

                throw ex;
            }
        }

        public DataSet GetRecordForReply(String activationcode)
        {
            try
            {
                DataSet dtResult = new DataSet();
                dbconn = new DatabaseConnection();
                dbconn.DatabaseConn("GetRecordForReply", "Main");

                if (dbconn.sqlConn.State == ConnectionState.Open) dbconn.sqlConn.Close();
                dbconn.sqlConn.Open();
                dbconn.sqlComm.CommandTimeout = 0;
                dbconn.sqlComm.Parameters.Clear();
                dbconn.sqlComm.Parameters.AddWithValue("@activationcode", activationcode);

                SqlDataAdapter da = new SqlDataAdapter();
                da.SelectCommand = dbconn.sqlComm;
                da.Fill(dtResult);
                dbconn.sqlComm.Connection.Close();
                dbconn.sqlConn.Close();
                return dtResult;
            }
            catch (Exception ex)
            {

                throw ex;
            }
        }
    }
}

File: BroadcastForBarangay/DLL Class/DAL/SMSDAL.vb
Public Class SMSDAL
    Private DbConn As New DatabaseConnection
    Private ActivationCodeValue As String

    Public Property ActivationCode As String
        Get
            Return ActivationCodeValue
        End Get
        Set(ByVal value As String)
            ActivationCodeValue = value.Trim
        End Set
    End Property
    Public Function InboxSave(ByVal SMSID As String, ByVal SMSContent As String, ByVal Sender As String, ByVal SMSDate As String, ByVal SMSTime As String, ByVal ModemIDValue As String) As String
        Dim message As String
        If DbConn.sqlconn.State = ConnectionState.Open Then DbConn.sqlconn.Close()
        DbConn.DatabaseConnection("InboxSave")
        message = ""
        Try
            With DbConn
                .sqlconn.Open()
                .sqlcomm.Parameters.Clear()
                .sqlcomm.Parameters.AddWithValue("@SMSID", SMSID)
                .sqlcomm.Parameters.AddWithValue("@SMSContent", SMSContent)
                .sqlcomm.Parameters.AddWithValue("@Sender", Sender)
                .sqlcomm.Parameters.AddWithValue("@SMSDate", SMSDate)
                .sqlcomm.Parameters.AddWithValue("@SMSTime", SMSTime)
                .sqlcomm.Parameters.AddWithValue("@ActivationCode", ActivationCode)
                .sqlcomm.Parameters.AddWithValue("@ModemID", ModemIDValue)
                message = .sqlcomm.ExecuteScalar()
                .sqlconn.Close()
            End With
            Return message
        Catch ex As Exception
            Throw ex
        End Try
    End Function
    Public Function SendMessage(ByVal Message As String, ByVal Sender As String) As Boolean
        'Dim message As String
        If DbConn.sqlconn.State = ConnectionState.Open Then DbConn.sqlconn.Close()
        DbConn.DatabaseConnection("SendMessage")
        Try
            With DbConn
                .sqlconn.Open()
                .sqlcomm.CommandTimeout = 0
                .sqlcomm.Parameters.Clear()
                .sqlcomm.Parameters.AddWithValue("@Message", Message)
                .sqlcomm.Parameters.AddWithValue("@SMSMobileNumber", Sender)
                Message = .sqlcomm.ExecuteScalar()
                .sqlconn.Close()
            End With
            Return True
        Catch ex As Exception
            Throw ex
        End Try
    End Function
    Public Function OutBoxSave(ByVal SMSContent As String, ByVal Recipient As String, ByVal Status As String, ByVal Remarks As String, ByVal InboxID As String, ByVal InboxType As String) As Boolean
        Dim message As Boolean
        If DbConn.sqlconn.State = ConnectionState.Open Then DbConn.sqlconn.Close()
        DbConn.DatabaseConnection("OutboxSave")
        message = False
        Try
            With DbConn
                .sqlconn.Open()
                .sqlcomm.Parameters.Clear()
                .sqlcomm.Parameters.AddWithValue("@SMSContent", SMSContent)
                .sqlcomm.Parameters.AddWithValue("@Recipient", Recipient)
                .sqlcomm.Parameters.AddWithValue("@Status", Status)
                .sqlcomm.Parameters.AddWithValue("@StatusRemarks", Remarks)
                .sqlcomm.Parameters.AddWithValue("@InboxID", InboxID)
                .sqlcomm.Parameters.AddWithValue("@InboxType", InboxType)
                .sqlcomm.ExecuteNonQuery()
                .sqlconn.Close()
                message = True
            End With
            Return message
        Catch ex As Exception
            Throw ex
        End Try
    End Function
    Public Function GetRecordForReply(ByVal activationcode As String, ByVal inboxID As String) As DataTable
        Dim dtResult As New DataTable
        If DbConn.sqlconn.State = ConnectionState.Open Then DbConn.sqlconn.Close()
        DbConn.DatabaseConnection("GetRecordForReply")
        Try
            With DbConn
                .sqlconn.Open()
                .sqlcomm.Parameters.Clear()
                .sqlcomm.Parameters.AddWithValue("@activationcode", activationcode)
                .sqlcomm.Parameters.AddWithValue("@inboxID", inboxID)

                Dim dr As SqlClient.SqlDataReader
                dr = .sqlcomm.ExecuteReader()
                dtResult.Load(dr)
                .sqlconn.Close()
            End With
            Return dtResult
        Catch ex As Exception
            Throw ex
        End Try
    End Function
End Class
